src/domain/position/fx-timeout-reserved.js

```diff
@@ -0,0 +1,153 @@
+const { Enum } = require('@mojaloop/central-services-shared')
+const ErrorHandler = require('@mojaloop/central-services-error-handling')
+const Config = require('../../lib/config')
+const Utility = require('@mojaloop/central-services-shared').Util
+const MLNumber = require('@mojaloop/ml-number')
+const Logger = require('@mojaloop/central-services-logger')
+
+/**
+ * @function processPositionFxTimeoutReservedBin
+ *
+ * @async
+ * @description This is the domain function to process a bin of timeout-reserved messages of a single participant account.
+ *
+ * @param {array} fxTimeoutReservedBins - an array containing timeout-reserved action bins
+ * @param {number} accumulatedPositionValue - value of position accumulated so far from previous bin processing
+ * @param {number} accumulatedPositionReservedValue - value of position reserved accumulated so far, not used but kept for consistency
+ * @param {object} accumulatedFxTransferStates - object with commitRequest id keys and fxTransfer state id values. Used to check if fxTransfer is in correct state for processing. Clone and update states for output.
+ * @param {object} transferInfoList - object with transfer id keys and transfer info values. Used to pass transfer info to domain function.
+ * @returns {object} - Returns an object containing accumulatedPositionValue, accumulatedPositionReservedValue, accumulatedTransferStateChanges, accumulatedFxTransferStates, resultMessages, limitAlarms or throws an error if failed
+ */
+const processPositionFxTimeoutReservedBin = async (
+  fxTimeoutReservedBins,
+  accumulatedPositionValue,
+  accumulatedPositionReservedValue,
+  accumulatedFxTransferStates,
+  fetchedReservedPositionChangesByCommitRequestIds
+) => {
+  const fxTransferStateChanges = []
+  const participantPositionChanges = []
+  const resultMessages = []
+  const accumulatedFxTransferStatesCopy = Object.assign({}, accumulatedFxTransferStates)
+  let runningPosition = new MLNumber(accumulatedPositionValue)
+  // Position action FX_RESERVED_TIMEOUT event messages are keyed with payer account id.
+  // We need to revert the payer's position for the source currency amount of the fxTransfer.
+  // We need to notify the payee of the timeout.
+  if (fxTimeoutReservedBins && fxTimeoutReservedBins.length > 0) {
+    for (const binItem of fxTimeoutReservedBins) {
+      Logger.isDebugEnabled && Logger.debug(`processPositionFxTimeoutReservedBin::binItem: ${JSON.stringify(binItem.message.value)}`)
+      const participantAccountId = binItem.message.key.toString()
+      const commitRequestId = binItem.message.value.content.uriParams.id
+      const fxp = binItem.message.value.to
+      const payerFsp = binItem.message.value.from
```

Comment:
Can we rename fxp as counterPartyFsp and payerFsp as initiatingFsp? Because 'from' is not always payer, it can be payee in payee side currency conversion.

Reply:
Yep! Mb. done.